packages/keyValueCache/src/InMemoryLRUCache.ts

```diff
@@ -0,0 +1,58 @@
+import LRUCache from "lru-cache";
+import type { KeyValueCache, KeyValueCacheSetOptions } from "./KeyValueCache";
+
+// LRUCache wrapper to implement the KeyValueCache interface.
+export class InMemoryLRUCache<T = string> implements KeyValueCache<T> {
+  private cache: LRUCache<string, T>;
+
+  constructor(lruCacheOpts?: LRUCache.Options<string, T>) {
+    this.cache = new LRUCache({
+      sizeCalculation,
+      // Create ~about~ a 30MiB cache by default. Configurable by providing
+      // `lruCacheOpts`.
+      maxSize: Math.pow(2, 20) * 30,
+      ...lruCacheOpts,
+    });
+  }
+
+  async set(key: string, value: T, options?: KeyValueCacheSetOptions) {
+    if (options?.ttl) {
+      this.cache.set(key, value, { ttl: options.ttl * 1000 });
+    } else {
+      this.cache.set(key, value);
+    }
+  }
+
+  async get(key: string) {
+    return this.cache.get(key);
+  }
+
+  async delete(key: string) {
+    return this.cache.delete(key);
+  }
+
+  clear() {
+    this.cache.clear();
+  }
+
+  keys() {
+    const keys = [];
+    for (const key of this.cache.keys()) {
+      keys.push(key);
+    }
+    return keys;
+  }
+}
+
+// default size calculator for strings and serializable objects, else naively
+// return 1
+export function sizeCalculation<T>(item: T) {
```

Comment:
This function name is a bit non-specific as a standalone function.  Maybe better as a static method on the class?

Reply:
Heh, yeah. I think this used to be that but ended up a separate function through some iteration around `Keyv`. Agree it makes sense taking this back to a static.